app/main.py

```diff
@@ -1,3 +1,42 @@
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int | float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: int | float) -> object:
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        return Distance(self.km + other)
+
+    def __iadd__(self, other: int | float) -> object:
+        if isinstance(other, Distance):
+            self.km += other.km
+        else:
+            self.km += other
```

Comment:
The `else` statement is unnecessary here. You can directly add `other` to `self.km` without the `else` block, as suggested in the checklist.